components/phpunit_bridge.rst

```diff
@@ -200,6 +200,26 @@ the ``vendors`` directory will make the test suite fail, while deprecations
 triggered from a library inside it will not, giving you the best of both
 worlds.
 
+Here is a summary that should help you pick the right mode:
+
++----------------+-----------------------------------------------------+
+| Mode           | Recommended situation                               |
++================+=====================================================+
+| strict         | Recommended for actively maintained projects        |
+|                | with little to no dependencies                      |
++----------------+-----------------------------------------------------+
+| <some integer> | Recommended for projects that you cannot            |
+|                | immediately fix but don't want to make worse.       |
+|                | Can be used to transition from one mode to another. |
++----------------+-----------------------------------------------------+
+| weak_vendors   | Recommended for libraries that use                  |
+|                | the deprecation system themselves and               |
+|                | cannot afford to use one of the modes above.        |
++----------------+-----------------------------------------------------+
+| weak           | Not recommended; will probably lead to              |
+|                | a big refactoring.                                  |
++----------------+-----------------------------------------------------+
```

Comment:
I like this new table! But, which mode should I use in my project? Most projects either:

A) Have deprecations & know about them... and just want to report them at the bottom of their tests (but not cause failures)

B) Do not have deprecations and want to keep it that way. New deprecations should cause the tests to fail.

What should I have for (A)? And is (B) `strict`? Or would `weak_vendors` be better (i.e. only tell me about *my* deprecations)?

Reply:
Right now, `weak` is not good enough for A, and I couldn't convince people to change this. At work, I observed that people are using `weak_vendors` to achieve A, although it's not meant for that. For B, you should use `strict` if you can. If you're a library and i turns out not to be realistic, for instance if the build fails on your contributors PRs because of unrelated changes in new minor versions of your dependencies, you might be better off with `weak_vendors`, which will let you focus on _your_ deprecations (`trigger_error` calls _you_ make in code _you_ should no longer call).